index.js

```diff
@@ -25,13 +25,31 @@ var UNICODE_CHARS = {
     '\u2029': '\\u2029'
 };
 
+// We can‘t just instanceof Date since dates are already converted to strings
+// because of native Date.prototype.JSON (which use toISOString)
+var DATE_LENGTH = new Date().toISOString().length
+function isDate(d) {
+  try {
+    return (
+      // testing length first to avoid new Date() for every string
+      d.length === DATE_LENGTH &&
```

Comment:
The date is automatically converted to ISOString by Date.prototype.toJSON before this code is executed. So this is not a problem. That's why I tested the length, because all date will be already converted to the same format at this point.